torch/lib/THC/generic/THCTensorMathBlas.cu

```diff
@@ -658,8 +658,20 @@ THC_API void THCTensor_(btrifact)(THCState *state, THCTensor *ra_, THCudaIntTens
 
   long num_batches = ra__->size[0];
 
-  THCudaIntTensor_resize2d(state, rpivots_, num_batches, n);
-  int *pivots_gpu = THCudaIntTensor_data(state, rpivots_);
+  if (!pivot) {
+    THCudaIntTensor *t = THCudaIntTensor_new(state);
+    THCudaIntTensor_range(state, t, 1, n, 1);
```

Comment:
Yes, I think (but am not sure) it's for backwards compatibility with code that uses Fortran. This is documented in [dgetrf](http://www.netlib.org/clapack/clapack-3.2.1-CMAKE/SRC/VARIANTS/lu/REC/dgetrf.c) as `IPIV` (Although as another note I am nearly certain that the description of the `IPIV` there is wrong.)